strongbox-storage/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/services/impl/ArtifactSearchServiceImpl.java

```diff
@@ -38,94 +52,166 @@
+                                    String repositoryId,
+                                    String pathToArtifactFile)
+    {
+        String baseUrl = getConfiguration().getBaseUrl();
+        baseUrl = (baseUrl.endsWith("/") ? baseUrl : baseUrl + "/");
+
+        return baseUrl + "storages/" + storageId + "/" + repositoryId + "/" + pathToArtifactFile;
+    }
+    
+    private List<SearchResult> indexSearch(String query,
+                                           Collection<Storage> storages)
+        throws ParseException,
+        IOException
+    {
+        List<SearchResult> result = new LinkedList<>();
+        for (Storage storage : storages)
         {
-            for (Storage storage : storages)
+            for (Repository r : storage.getRepositories().values())
             {
-                for (Repository r : storage.getRepositories().values())
+                logger.debug("Repository: {}", r.getId());
+                final RepositoryIndexer repositoryIndex = repositoryIndexManager.getRepositoryIndexer(getIndexId(storage, r));
+                if (repositoryIndex != null)
                 {
-                    logger.debug("Repository: {}", r.getId());
-
-                    String contextId = storage.getId() + ":" + r.getId() + ":" + IndexTypeEnum.LOCAL.getType();
-                    final RepositoryIndexer repositoryIndexer = repositoryIndexManager.getRepositoryIndexer(contextId);
-                    if (repositoryIndexer != null)
+                    final Set<SearchResult> sr = repositoryIndex.search(query);
+                    if (sr != null && !sr.isEmpty())
                     {
-                        final Set<SearchResult> sr = repositoryIndexer.search(searchRequest.getQuery());
-
-                        if (sr != null && !sr.isEmpty())
-                        {
-                            searchResults.getResults().addAll(sr);
-                        }
+                        result.addAll(sr);
                     }
                 }
             }
+        }
+        return result;
+    }
 
-            logger.debug("Results: {}", searchResults.getResults().size());
+    private List<SearchResult> indexSearch(String query,
+                                           Storage storage,
+                                           String repositoryId)
+        throws ParseException,
+        IOException
+    {
+        Repository repository = storage.getRepository(repositoryId);
+        String storageAndRepositoryId = getIndexId(storage, repository);
+        List<SearchResult> result = new LinkedList<>();
+        final Set<SearchResult> sr = repositoryIndexManager.getRepositoryIndexer(storageAndRepositoryId)
+                                                           .search(query);
+        if (!sr.isEmpty())
+        {
+            result.addAll(sr);
+        }
+        return result;
+    }
 
-            return searchResults;
+    protected String getIndexId(Storage storage,
```

Comment:
Across the codebase, the is actually referred to as the `contextId` of the index. Could you please rename this?